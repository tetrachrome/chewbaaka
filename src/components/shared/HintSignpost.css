/**
 * HintSignpost.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 07, 2020
 * Updated  : Jul 07, 2020
 */

div.HintSignpostIconContainer {
  float: right;
  /** Icon is too close to label text on the left. Leave some room. */
  margin-left: 5px;
}
