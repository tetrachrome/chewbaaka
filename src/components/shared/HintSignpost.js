/**
 * HintSignpost.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 07, 2020
 * Updated  : Jul 07, 2020
 */

 /**
 * HintSignpost
 *
 * `HintSignpost` is a component that illustrates
 * a specified text string as a hint message.
 *
 * This component/widget is intended to demonstrate
 * suggested actions or cues to users.
 *
 * Props:
 *
 *  - `hintText`: Text of the hint message.
 */

import React from 'react';

import { Icon, Label } from 'semantic-ui-react'

import './HintSignpost.css'

function HintSignpost(props) {
  return (
    <div className="HintSignpostOuterContainer">
      <div className="HintSignpostInnerContainer">
        <LabelÂ size="large" color="blue">
          {props.hintText}
          <div className="HintSignpostIconContainer">
            <Icon name="arrow circle down" size="large"/>
          </div>
        </Label>
      </div>
    </div>
  );
}

export default HintSignpost;
