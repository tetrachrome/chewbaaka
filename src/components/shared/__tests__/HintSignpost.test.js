/**
 * HintSignpost.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 07, 2020
 * Updated  : Jul 08, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import HintSignpost from '../HintSignpost';

const hintText = "Hover over images below to reveal the genres";

test('renders HintSignpost component', () => {
  render(
    <HintSignpost hintText={hintText} />
  );

  const labelElement = screen.getByText(hintText);
  expect(labelElement).toBeInTheDocument();
});

test('HintSignpost component snapshot', () => {
  const tree = renderer
    .create(
      <HintSignpost hintText={hintText} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
