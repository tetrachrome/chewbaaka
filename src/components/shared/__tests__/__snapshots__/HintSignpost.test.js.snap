// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintSignpost component snapshot 1`] = `
<div
  className="HintSignpostOuterContainer"
>
  <div
    className="HintSignpostInnerContainer"
  >
    <div
      className="ui blue large label"
      onClick={[Function]}
    >
      Hover over images below to reveal the genres
      <div
        className="HintSignpostIconContainer"
      >
        <i
          aria-hidden="true"
          className="lightbulb outline large icon"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
